deps/libgit2.mk

```diff
@@ -45,6 +47,10 @@ ifeq ($(OS),Linux)
 $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/libgit2-require-openssl.patch-applied $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/libgit2-openssl-hang.patch-applied
 endif
 $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/CMakeLists.txt $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/libgit2-ssh.patch-applied $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/libgit2-agent-nonfatal.patch-applied $(LIBSSH2_OBJ_TARGET)
+ifneq ($(OS),WINNT)
+$(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(CURL_OBJ_TARGET)
+endif
+$(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile:
```

Comment:
ordering is determined by whether or not something is a dependency. if you need curl after libssh2, then libssh2 needs to be a dependency of libcurl. within the normal dependencies of a single target, the order shouldn't matter. https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html